### PR TITLE
Update 2k

### PR DESCRIPTION
**Pull request by rmariani**
Adding the missing entry `+-nip:to`, added discussion sections, filled in missing bodies of entries, updated source code, updated syntax, added examples

Thanks to Ted for helping with the discussion of the tree.

### CODE REVIEW

File: docs/hoon/library/2k.md [hunk 1]
```diff
@@ -8,38 +8,41 @@ navhome: /docs/
...
+#### Source
 
-    ++  to                                                  ::  queue engine
-      |/  a/(qeu)
+    ++  to                                                 
+      |_  a=(tree)
 
+#### Discussion
+
+`tree` is the generic tree-shape mold. Maps, sets, and queues all share that
```
Comment (belisarius222) [inline docs/hoon/library/2k.md]: `+tree` is the generic mold builder for trees where each node looks like either `~` or `[n=<some value> l=tree r=tree]`.

File: docs/hoon/library/2k.md [hunk 2]
```diff
@@ -8,38 +8,41 @@ navhome: /docs/
...
+shape, but they put their elements in different places and have different
```
Comment (belisarius222) [inline docs/hoon/library/2k.md]: The places where the elements can go are the same; they're just ordered differently. I'd say something more like:
`but they store their elements in different orders and have different interfaces`.

File: docs/hoon/library/2k.md [hunk 3]
```diff
@@ -49,128 +52,119 @@ Source
...
+    > =a (~(gas to `(qeu @)`~) [1 2 3 ~])
+    > =b [4 5 6 ~]
+    > `(qeu)`(~(gas to a) b)
+    {6 5 4 3 2 1}
 
+***
 ### `+-get:to`
 
-Pop
+Head-tail pair
```
Comment (belisarius222) [inline docs/hoon/library/2k.md]: I don't understand what 'head' means here. I assume it means the element that was inserted first? I think we should document that this is a FIFO queue, as opposed to some other kind of queue.

File: docs/hoon/library/2k.md [hunk 4]
```diff
@@ -180,164 +174,192 @@ Source
...
-    > a
+
+    > `(qeu)`a
     {6 5 4 3 2 1}
 
-------------------------------------------------------------------------
+***
+###  `+-nip:to`
+
+Removes the root of queue `a`, producing the resulting queue.
```
Comment (belisarius222) [inline docs/hoon/library/2k.md]: What significance does the root element have? I wonder if this is really just part of the private interface for the queue to rebalance itself, as opposed to something intended to be called publicly. If so, we should mention that in the documentation.

File: docs/hoon/library/2k.md [hunk 5]
```diff
@@ -180,164 +174,192 @@ Source
...
 
-------------------------------------------------------------------------
-
+***
 ### `+-tap:to`
 
 Queue to list
 
-Produces `++queue` `a` as a `++list` from front to back.
+Produces queue `a` as a list from front to back.
```
Comment (belisarius222) [inline docs/hoon/library/2k.md]: Does this mean it produces the elements in FIFO order, or the reverse? the "front" and "back" concepts don't seem totally clear to me.

File: docs/hoon/library/2k.md [hunk 6]
```diff
@@ -180,164 +174,192 @@ Source
...
 
 A queue.
 
-Source
-------
+#### Source
 
-      +-  nap                                               ::  removes head
+      +-  nap                                              
         ?>  ?=(^ a)
```
Comment (belisarius222) [inline docs/hoon/library/2k.md]: Looks like this errors out on an empty queue. That should be documented.

Comment (rmariani) [reply docs/hoon/library/2k.md]: Interesting, will do so

**Comment by msutherl** [comment]
How's this PR doing?

**Comment by rmariani** [comment]
I'm going to finally go through it and make it Good

**Pull request closed without merging**